flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -2136,6 +2142,86 @@ IntrinsicLibrary::genSum(mlir::Type resultType,
                       "unexpected result for Sum", args);
 }
 
+// TRANSFER
+fir::ExtendedValue
+IntrinsicLibrary::genTransfer(mlir::Type resultType,
+                              llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  assert(args.size() >= 2); // args.size() == 2 when size argument is omitted.
+
+  // Handle source argument
+  auto source = builder.createBox(loc, args[0]);
+
+  // Handle mold argument
+  auto mold = builder.createBox(loc, args[1]);
+  fir::BoxValue moldTmp = builder.createBox(loc, args[1]);
```

Comment:
`createBox` creates IR, even though unused IR will be removed by dead code elimination, when it is rather trivial to avoid generating dead code, it's best to avoid it.
```
auto mold = builder.createBox(loc, args[1]);
fir::BoxValue moldTmp = mold;
```
Should get you the values you need without generating the IR twice.